File: PrintTreeFromTopToBottom/PrintTreeFromTopToBottom.cpp
// C++ Solution 1:
class Solution {
public:
    vector<int> PrintFromTopToBottom(TreeNode* root) {
        vector<int> res;
        if (root==NULL)
            return res;
        queue<TreeNode*> q;
        q.push(root);
        while(!q.empty())
        {
            TreeNode* temp = q.front();
            q.pop();
            res.push_back(temp->val);
            if (temp->left)
                q.push(temp->left);
            if (temp->right)
                q.push(temp->right);
        }
        return res;
    }
};
// C++ Solution 2:
class Solution {
public:
    vector<int> PrintFromTopToBottom(TreeNode* root) {
        vector<int> res;
        if (root==NULL)
            return res;
        deque<TreeNode*> q;
        q.push_back(root);
        while(q.size())
        {
            TreeNode* temp = q.front();
            q.pop_front();
            res.push_back(temp->val);
            if (temp->left)
                q.push_back(temp->left);
            if (temp->right)
                q.push_back(temp->right);
        }
        return res;
    }
};

File: PrintTreeFromTopToBottom/PrintTreeFromTopToBottom.py
class Solution:
    # 返回从上到下每个节点值列表，例：[1,2,3]
    def PrintFromTopToBottom(self, root):
        # write code here
        res = []
        if root is None:
            return res
        q = []
        q.append(root)
        while len(q)>0:
            temp = q.pop(0)
            res.append(temp.val)
            if temp.left:
                q.append(temp.left)
            if temp.right:
                q.append(temp.right)
        return res
## collections.deque()
class Solution:
    # 返回从上到下每个节点值列表，例：[1,2,3]
    def PrintFromTopToBottom(self, root):
        # write code here
        res = []
        if root is None:
            return res
        q = collections.deque()
        q.append(root)
        while q:
            node = q.popleft()
            res.append(node.val)
            if node.left:
                q.append(node.left)
            if node.right:
                q.append(node.right)
        return res
